scripts/upgrade

```diff
@@ -3,314 +3,154 @@
-            upstream_version="${version}0"
-            break
-        fi
-    done
-fi
-
-restart_gitea() {
-    # Set permissions
-    set_permission
-    ynh_systemd_action --service_name=$app --action="start" --log_path="/var/log/$app/gitea.log" --line_match="Starting new Web server: tcp:127.0.0.1:"
-    # Leave the time to update the database schema
+_run_gitea_migration() {
+    source_id="$1"
+    ynh_setup_source --dest_dir="$install_dir" --source_id="$source_id"
+    chmod +x "$install_dir/gitea"
+    _gitea_permissions_install_dir
+    ynh_systemd_action --service_name="$app" --action="start" --log_path="/var/log/$app/gitea.log" --line_match="Starting new Web server: tcp:127.0.0.1:"
+    #FIXME: Leave the time to update the database schema
     sleep 5
-    systemctl stop $app
+    systemctl stop "$app"
 }
 
-case $upstream_version in
-"0.0."* )
-    ynh_setup_source $final_path source/${architecture}_1.0
-    set_permission
-    systemctl start $app
-    sleep 20
-    systemctl stop $app
-;&
-"1.0."* )
-    ynh_setup_source $final_path source/${architecture}_1.1
-    restart_gitea
-;&
-"1.1."* )
-    ynh_setup_source $final_path source/${architecture}_1.2
-    restart_gitea
-;&
-"1.2."* )
-    ynh_setup_source $final_path source/${architecture}_1.3
-    restart_gitea
-;&
-"1.3."* )
-    ynh_setup_source $final_path source/${architecture}_1.4
-    restart_gitea
-;&
-"1.4."* )
-    ynh_setup_source $final_path source/${architecture}_1.5
-    restart_gitea
-;&
-"1.5."* )
-    ynh_setup_source $final_path source/${architecture}_1.6
-    restart_gitea
-;&
-"1.6."* )
-    ynh_setup_source $final_path source/${architecture}_1.7
-    restart_gitea
-;&
-"1.7."* )
-    ynh_setup_source $final_path source/${architecture}_1.8
-    restart_gitea
-;&
-"1.8."* )
-    ynh_setup_source $final_path source/${architecture}_1.9
-    restart_gitea
-;&
-"1.9."* )
-    ynh_setup_source $final_path source/${architecture}_1.10
-    restart_gitea
-;&
-"1.10."* )
-    ynh_setup_source $final_path source/${architecture}_1.11
-    restart_gitea
-;&
-"1.11."* )
-    ynh_setup_source $final_path source/${architecture}_1.12
-    restart_gitea
-;&
-"1.12."* )
-    ynh_setup_source $final_path source/${architecture}_1.13
-    restart_gitea
-;&
-"1.13."* )
-    ynh_setup_source $final_path source/${architecture}_1.14
-    restart_gitea
-;&
-"1.14."* )
-    ynh_setup_source $final_path source/${architecture}_1.15
-    restart_gitea
-;&
-"1.15."* )
-    ynh_setup_source $final_path source/${architecture}_1.16
-    restart_gitea
-;&
-"1.16."* )
-    ynh_setup_source $final_path source/${architecture}_1.17
-    restart_gitea
-;&
-"1.17."* )
-    ynh_setup_source $final_path source/${architecture}_1.18
-    restart_gitea
-;&
-"1.18."* )
-    ynh_setup_source $final_path source/${architecture}_1.19
-    restart_gitea
-;&
-esac
+if ynh_compare_current_package_version --comparison lt --version "1.6.4~ynh1"; then
```

Comment:
Just a question: why do you do DB migration only if the version < 1.6 ?

Reply:
Hi,
I’ve exchanged with the Gitea devs about DB migration. They explained we just need to go up to 1.6.4 step by step, then we can "long jump" to the latest version ; the intermediate migration steps are now kept in the new versions.

Reply:
I’m not sure but I think it’s there : https://github.com/go-gitea/gitea/tree/main/models/migrations

Reply:
Hello, thanks for the answer. In this case we could maybe just remove the support of upgrade from version <1.6 it could bring many simplification on upgrade. I think the 1.6 version is quite old it should be ok.

Reply:
If you’re OK with that, I’m all for it ! Maybe let’s print an error message to tell to run `yunohost app upgrade gitea -u https://github.com/yunohost-apps/gitea_ynh/commit/SOME_SHA_BEFORE_THIS_MERGE_REQUEST` for people on old versions :)